lib/compiler/src/beam_ssa_bsm.erl

```diff
@@ -450,21 +450,22 @@ combine_matches({Fs0, ModInfo}) ->
 combine_matches(#b_function{bs=Blocks0,cnt=Counter0}=F, ModInfo) ->
     case funcinfo_get(F, has_bsm_ops, ModInfo) of
         true ->
-            RPO = beam_ssa:rpo(Blocks0),
-            {Dominators, _} = beam_ssa:dominators(RPO, Blocks0),
+            RPO0 = beam_ssa:rpo(Blocks0),
+            {Dominators, _} = beam_ssa:dominators(RPO0, Blocks0),
             {Blocks1, State} =
                 beam_ssa:mapfold_blocks(
                   fun(Lbl, #b_blk{is=Is0}=Block0, State0) ->
                           {Is, State} = cm_1(Is0, [], Lbl, State0),
                           {Block0#b_blk{is=Is}, State}
                   end,
-                  RPO,
-                  #cm{ definitions = beam_ssa:definitions(Blocks0),
+                  RPO0,
+                  #cm{ definitions = beam_ssa:definitions(RPO0, Blocks0),
                        dominators = Dominators,
                        blocks = Blocks0 },
                   Blocks0),
 
-            Blocks2 = beam_ssa:rename_vars(State#cm.renames, [0], Blocks1),
+            RPO1 = beam_ssa:rpo(Blocks1),
```

Comment:
You can reuse the previous RPO here, because RPO0 and RPO1 are guaranteed to be identical (the fun passed to `beam_ssa:mapfold_blocks/4` does not update terminators).

Reply:
Fantastic, I will update this and leave a comment explaining why. :)

Reply:
Done!